File: Easy/Valid Anagram/validanagram.java
import java.util.HashMap;

class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() !=  t.length()){
            return false; //if the lengths of the two strings do not match, they cannot be anagrams of each other
        }
        HashMap<Character,Integer> letters = new HashMap<Character,Integer>();
        for (int i = 0; i < s.length(); i++){
            if (!letters.containsKey(s.charAt(i))) letters.put(s.charAt(i),1);
            else letters.put(s.charAt(i),letters.get(s.charAt(i))+1);
        }
        for (int i = 0; i < t.length(); i++){
            if (!letters.containsKey(t.charAt(i))) return false;
            letters.put(t.charAt(i),letters.get(t.charAt(i))-1);
        }
        for (char k: letters.keySet()){
            if (letters.get(k) != 0) return false;
        }
        return true;
    }
}

File: Easy/Valid Anagram/validanagram.py
class Solution:
    #O(n) where n is length of both strings if both strings are equal in length
    def isAnagram(self, s: str, t: str) -> bool: 
        if len(s) != len(t): #if length of the strings are different, they cannot be anagrams
            return False
        letters = {}
        for char in s:
            letters[char] = 1 if char not in letters else letters[char] + 1
        for char in t:
            if char in letters:
                letters[char] -= 1
            else:
                return False
        for key in letters:
            if letters[key] != 0:
                return False
        return True
